### PR TITLE
add I/O info for async functions

### LINKED ISSUE #6971: [BUG] async clients cannot do i/o (parquet write)
**What happened**:

async client to_parquet throws an exn on simple files. 

This bug report is a continuation of the async client read_json bug report in https://github.com/dask/dask/issues/6961, with a similar error. Both might  be cases of https://github.com/dask/dask/issues/5518 . Put together, I'm wondering if the broader issue is async client is incompatible with normal i/o methods like reading/writing data via dd?

**What you expected to happen**:

async client can successfully run `to_parquet`

**Minimal Complete Verifiable Example**:

Except to write with no exceptions, but surprisingly, fails:

```python
import dask, dask.dataframe as dd, pandas as pd
from dask.distributed import Client, LocalCluster

async def test_dask_write_parquet_async():
    df = pd.DataFrame({'x': [1, 2]})
    async with await LocalCluster(asynchronous=True, dashboard_address=None, worker_dashboard_address=None) as cluster:
        async with await Client(cluster, asynchronous=True) as client:
            #dask.config.set(scheduler=client)
            ddf = dd.from_pandas(df, chunksize=1000000)
            ddf.to_parquet('/tmp/some.parquet', write_index=False)
```

=>

```
test/server/util/test_dask.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py:3974: in to_parquet
    return to_parquet(self, path, *args, **kwargs)
/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/io/parquet/core.py:508: in to_parquet
    out = out.compute(**compute_kwargs)
/conda/envs/rapids/lib/python3.7/site-packages/dask/base.py:167: in compute
    (result,) = compute(self, traverse=False, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (Delayed('write_metadata-eb167a5d-6987-45e4-88ea-9b7c66ddc2b7'),), kwargs = {}, traverse = False, optimize_graph = True, collections = [Delayed('write_metadata-eb167a5d-6987-45e4-88ea-9b7c66ddc2b7')]
repack = <function unpack_collections.<locals>.repack at 0x7efd2a1db4d0>, schedule = <bound method Client.get of <Client: not connected>>
dsk = {'write_metadata-eb167a5d-6987-45e4-88ea-9b7c66ddc2b7': (<function apply at 0x7efd2d2f1680>, <function ArrowEngine.wri..., 'from_pandas-6eca1cbba36fc5731fae4cac36111fc4'), ('from_pandas-6eca1cbba36fc5731fae4cac36111fc4', 0):    x
0  1
1  2}
keys = [['write_metadata-eb167a5d-6987-45e4-88ea-9b7c66ddc2b7']], postcomputes = [(<function single_key at 0x7efd2cdc7830>, ())], x = Delayed('write_metadata-eb167a5d-6987-45e4-88ea-9b7c66ddc2b7')
results = <coroutine object Client._gather at 0x7efd2a1db290>

    def compute(*args, **kwargs):
        """Compute several dask collections at once.
...
        dsk = collections_to_dsk(collections, optimize_graph, **kwargs)
        keys, postcomputes = [], []
        for x in collections:
            keys.append(x.__dask_keys__())
            postcomputes.append(x.__dask_postcompute__())
    
        results = schedule(dsk, keys, **kwargs)
>       return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
E       TypeError: zip argument #1 must support iteration

/conda/envs/rapids/lib/python3.7/site-packages/dask/base.py:453: TypeError
============================================================================================= short test summary info =============================================================================================
FAILED test/server/util/test_dask.py::test_dask_write_parquet_async - TypeError: zip argument #1 must support iteration
```

**Anything else we need to know?**:

Sync worked:

```python
async def test_dask_write_parquet_sync():
    df = pd.DataFrame({'x': [1, 2]})
    with LocalCluster(dashboard_address=None, worker_dashboard_address=None) as cluster:
        with Client(cluster) as client:
            #dask.config.set(scheduler=client)
            ddf = dd.from_pandas(df, chunksize=1000000)
            ddf.to_parquet('/tmp/some.parquet', write_index=False)
```

**Environment**:

docker -> conda -> python 3.7.9 -> rapids 0.17 (stable) -> dask 2.30.0

### PR DESCRIPTION
**Pull request by lognorman20**
- [x] Closes #6971 
- [x] Passes `pre-commit run --all-files`
- [x] Tests added / passed

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by ncclementi** [comment]
@charlesbluca would you mind doing your magic here to get the gpuCI running?

File: docs/source/futures.rst [hunk 1]
```diff
@@ -872,6 +872,12 @@ All operations that require talking to the remote worker are awaitable:
 
        n = await counter.n  # attribute access also must be awaited
 
+Generally, all I/O operations should be done using the :code:`compute=False` 
+parameter to avoid asynchronous blocking:
+
+.. code-block:: python
+
+   df3 = await client.compute(dd.read_parquet('/tmp/some.parquet', compute=False))
```
Comment (ncclementi) [inline docs/source/futures.rst:880]: [`dd.read_parquet`](https://docs.dask.org/en/stable/generated/dask.dataframe.read_parquet.html) doesn't have a `compute` option. I think this example was supposed to be for [`dd.to_parquet`](https://docs.dask.org/en/stable/generated/dask.dataframe.to_parquet.html) by looking at the discussions in https://github.com/dask/dask/issues/6971

```suggestion
   df3 = await client.compute(dd.to_parquet('/tmp/some.parquet', compute=False))
```

File: docs/source/futures.rst [hunk 2]
```diff
@@ -872,6 +872,12 @@ All operations that require talking to the remote worker are awaitable:
 
        n = await counter.n  # attribute access also must be awaited
 
+Generally, all I/O operations should be done using the :code:`compute=False`
```
Comment (ncclementi) [inline docs/source/futures.rst:875]:
```suggestion
Generally, all I/O operations that trigger computations (e.g. ``to_parquet``) should be done using the ``compute=False`` 
```

**Review by ncclementi — COMMENTED** [review]
Leaving some small comments here, but this looks good!! 
@jsignell since you were involved originally in https://github.com/dask/dask/issues/6971 would you mind giving it a second look

**Comment by ncclementi** [comment]
It seems this PR addresses @jsignell suggestion https://github.com/dask/dask/issues/6971#issuecomment-748092054 
but I'm not sure this closes the issue. Or is it implicit that a fix in the docs is what we can do here?

**Comment by charlesbluca** [comment]
add to allowlist

**Comment by lognorman20** [comment]
@ncclementi just updated with respect to your suggestions

**Comment by jsignell** [comment]
This looks good to me!

File: docs/source/futures.rst [hunk 3]
```diff
@@ -872,6 +872,12 @@ All operations that require talking to the remote worker are awaitable:
 
        n = await counter.n  # attribute access also must be awaited
 
+Generally, all I/O operations that trigger computations (e.g. ``to_parquet``) should be done using the ``compute=False`` 
+parameter to avoid asynchronous blocking:
+
+.. code-block:: python
+
+   df3 = await client.compute(dd.to_parquet('/tmp/some.parquet', compute=False))
```
Comment (jrbourbeau) [inline docs/source/futures.rst:880]: Apologies for the delayed review. A couple of comments:

- I think the result of this computation is `None`. Assigning it a variable name of `df3`, while technically valid, is a bit confusing as it makes me think it's a `DataFrame`
- The `dd.to_parquet(...)` signature needs a `DataFrame` passed as the first argument

Comment (jsignell) [reply docs/source/futures.rst:880]: I pushed the green button too fast. I'll fix this up.

**Pull request merged**